source/plugins/ruby/out_mdm.rb

```diff
@@ -98,7 +99,20 @@ def start
           if aks_resource_id.downcase.include?("microsoft.kubernetes/connectedclusters")
             @isArcK8sCluster = true
           end
-          @@post_request_url = @@post_request_url_template % { aks_region: aks_region, aks_resource_id: aks_resource_id }
+
+          # If Custom_Metrics_Endpoint provided, the url format shall be validated before emitting metrics into given endpoint.
+          Custom_Metrics_Endpoint = ENV['Custom_Metrics_Endpoint']
```

Comment:
change the variable naming convention to custom_metrics_endpoint to be consistent with other local variables.